src/nxt_http_static.c

```diff
@@ -879,12 +879,7 @@ nxt_http_static_buf_completion(nxt_task_t *task, void *obj, void *data)
 
     n = nxt_file_read(fb->file, b->mem.start, size, fb->file_pos);
 
-    if (n != size) {
-        if (n >= 0) {
-            nxt_log(task, NXT_LOG_ERR, "file \"%FN\" has changed "
-                    "while sending response to a client", fb->file->name);
-        }
-
+    if (n == -1) {
```

Comment:
It probably makes sense to wrap this check using `nxt_slow_path` and add a debug message that prints the error code. Message could be also helpful in identifying the place where `nxt_http_request_error_handler` was called from.

Reply:
The error message is already printed in nxt_file_read().

Reply:
Indeed. In this case, I would suggest checking `n` against `NXT_ERROR` (not `-1`) sinse `nxt_file_read` can return only `n >= 0` or `NXT_ERROR`.